src/components/fields/location-field/location-modal/location-modal.tsx

```diff
@@ -0,0 +1,423 @@
+	// RENDER FUNCTIONS
+	// =============================================================================
+	const renderNetworkErrorPrompt = () => {
+		/**
+		 * Do not render any other error if there is no internet connectivity
+		 * since the form is not interactive.
+		 * When network restored, the form value will used.
+		 */
+		if (!hasInternetConnectivity || !showLocationModal) return;
+
+		if (showOneMapError) {
+			return (
+				<Prompt
+					id={TestHelper.generateId(id, "onemap-error")}
+					data-testid={TestHelper.generateId(id, "onemap-error")}
+					title="Map not available"
+					size="large"
+					show={true}
+					image={<ErrorImage src={ERROR_SVG} />}
+					description={
+						<Description weight="regular">
+							Sorry, there was a problem with the map. You&rsquo;ll not be able to enter the location
+							right now. Please try again later.
+							<br />
+							<br />
+							Do note that you&rsquo;ll not be able to submit your report without entering the location.
+						</Description>
+					}
+					buttons={[
+						{
+							id: "ok",
+							title: "OK",
+							onClick: () => {
+								setShowOneMapError(false);
+								handleCloseLocationModal();
+							},
+						},
+					]}
+				/>
+			);
+		}
+
+		if (showGetLocationError) {
+			return (
+				<Prompt
+					id={TestHelper.generateId(id, "get-location-error")}
+					data-testid={TestHelper.generateId(id, "get-location-error")}
+					title="Enable location settings"
+					size="large"
+					show={true}
+					description={
+						"We need your permission to determine your location. Enable location access in your browser and device settings, or enter your location manually."
+					}
+					buttons={[
+						{
+							id: "ok",
+							title: "OK",
+							onClick: handleCloseLocationPermissionModal,
+						},
+					]}
+				/>
+			);
+		}
+
+		if (showGetLocationTimeoutError) {
+			return (
+				<Prompt
+					id={TestHelper.generateId(id, "get-location-timeout-error")}
+					data-testid={TestHelper.generateId(id, "get-location-timeout-error")}
+					title="Something went wrong"
+					size="large"
+					show={true}
+					image={<ErrorImage src={TIMEOUT_SVG} />}
+					description={
+						<Description weight="regular">
+							It&rsquo;s taking longer than expected to retrieve your location. Please exit the map and
+							try again.
+						</Description>
+					}
+					buttons={[
+						{
+							id: "ok",
+							title: "OK",
+							onClick: () => {
+								setShowGetLocationTimeoutError(false);
+								handleCloseLocationModal();
+							},
+						},
+					]}
+				/>
+			);
+		}
+	};
+
+	return (
+		<>
+			<PrefetchImage src={OFFLINE_IMAGE} alt="no internet connectivity" />
+			<Modal
+				id={TestHelper.generateId(id, "modal", showLocationModal ? "show" : "hide")}
+				show={showLocationModal}
+			>
+				<ModalBox id={TestHelper.generateId(id, "box")} showCloseButton={false} mastheadHeight={mastheadHeight}>
```

Comment:
exposing `mastheadHeight` to the schema can be confusing because it is only applicable for mobile layouts and its use is not immediately apparent.

instead of this, you can remove this prop and assign a `className` to `ModalBox`, the class name can be derived from the schema like `${className}-modal-box`. developers will customise the masthead height by setting the top padding in the modal box instead.